src/main/scala/sigmastate/lang/SigmaPredef.scala

```diff
@@ -15,16 +15,72 @@ object SigmaPredef {
     declaration: Lambda,
     /** Builder of SigmaIR node which is equivalent to function application
       * Rule: Apply(f, args) -->  irBuilder(f, args) */
-    irBuilder: (SValue, Seq[SValue]) => SValue
-  )
+    irBuilder: PartialFunction[(SValue, Seq[SValue]), SValue]) {
```

Comment:
@ergomorphic What do you think of making `irBuilder` a `PartialFunction`? That makes function declaration succinct and much more readable with matching for arguments.

Reply:
@greenhat semantically it is not partial. For code readability you can use some helper which will transform PartialFunction to Function with error in default case.

Reply:
@greenhat Do you use this partial function for pattern matching?